src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+
+  actions.forEach((action) => {
+    const previousState =
+      states.length > 0 ? states[states.length - 1] : { ...state };
+    let newState = { ...previousState };
```

Comment:
[NAMING] - The variable name 'newState' is a bit generic. A more descriptive name like 'stateClone' or 'currentState' would better convey the purpose of this variable.